File: 771/solution.go
package _771

func numJewelsInStones(jewels string, stones string) int {
	dict := map[rune]bool{}
	for _, c := range jewels {
		dict[c] = true
	}
	res := 0
	for _, c := range stones {
		_, ok := dict[c]
		if ok == true {
			res++
		}
	}
	return res
}

File: 771/Solution.java
import java.util.HashSet;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashSet<Character> hashSet = new HashSet<>();
        for (int i = 0; i < jewels.length(); i++) {
            hashSet.add(jewels.charAt(i));
        }
        int res = 0;
        for (int i = 0; i < stones.length(); i++) {
            if (hashSet.contains(stones.charAt(i))) {
                res++;
            }
        }
        return res;
    }
}